lib/eventFactory.js

```diff
@@ -236,15 +234,36 @@ function createBuilds(config) {
         return null;
     }
 
-    // enable pipeline parameters feature
-    if (pipeline.parameters && pipelineConfig.meta && pipelineConfig.meta.parameters) {
-        const allowedPipelineParameters = _.keys(pipeline.parameters);
-        const currentPipelineParameters = _.assign({},
-            pipeline.parameters,
-            pipelineConfig.meta.parameters);
+    // enable pipeline parameters feature, if parameters exist in screwdriver.yaml
+    if (pipeline.parameters) {
+        const allowedParameters = Object.create(null);
+
+        Object.entries(pipeline.parameters).forEach(([name, val]) => {
+            let defaultParameterValue = val;
+
+            if (typeof val === 'object') {
+                defaultParameterValue = val.value;
+            }
+
+            allowedParameters[name] = { value: defaultParameterValue };
+            if (pipelineConfig.meta.parameters) {
+                let actualParameterValue = pipelineConfig.meta.parameters[name];
+
+                if (typeof actualParameterValue === 'object') {
+                    actualParameterValue = actualParameterValue.value;
+                }
+
+                if (actualParameterValue !== undefined) {
+                    allowedParameters[name] = { value: actualParameterValue };
+                }
+            }
+        });
 
-        pipelineConfig.meta.parameters = _.pick(currentPipelineParameters,
-            allowedPipelineParameters);
+        if (pipelineConfig.meta) {
```

Comment:
Can just do 
```
pipelineConfig.meta = Object.assign({}, pipelineConfig.meta, { parameters: allowedParameters });
```

so you don't need if/else block here

Reply:
I actually went back and forth on it, and eventually pick readability over shorter codes. 🤦‍♂️